plugins/discourse-narrative-bot/plugin.rb

```diff
@@ -227,5 +227,17 @@ module ::DiscourseNarrativeBot
     end
   end
 
+  self.on(:site_setting_changed) do |name, old_value, new_value|
+    next if name.to_s != "default_locale"
+    next if !SiteSetting.discourse_narrative_bot_enabled
+
+    profile = UserProfile.find_by(user_id: DiscourseNarrativeBot::BOT_USER_ID)
+
+    next if profile.blank?
+
+    new_bio = I18n.with_locale(new_value) { I18n.t("discourse_narrative_bot.bio") }
+    profile.update!(bio_raw: new_bio) if profile.bio_raw != new_bio
```

Comment:
Extremely minor, but I wonder if AR will prevent this kind of noop before reaching out to the DB?

Reply:
O good catch! AR actually does a noop.

```
[5] pry(main)> profile.update!(bio_raw: profile.bio_raw)
  TRANSACTION (0.2ms)  BEGIN
  User Load (4.1ms)  SELECT "users".* FROM "users" WHERE "users"."id" = -2 LIMIT 1
  TRANSACTION (0.2ms)  COMMIT
=> true
```